get_spice.py

```diff
@@ -257,19 +257,35 @@ def get_cspice():
     @staticmethod
     def check_for_spice():
         print("Checking the path", cspice_dir)
-        if not os.path.exists(cspice_dir):
+        if os.path.exists(cspice_dir):
+            return False
 
-            message = "Unable to find CSPICE at {0}. Attempting to Download CSPICE For you:".format(
-                cspice_dir
-            )
-            print(message)
-            # Download cspice using getspice.py
-            GetCSPICE(version="N0066")
-            if not os.path.exists(cspice_dir):
-                message = "Unable to find CSPICE at {0}. Exiting".format(cspice_dir)
-                sys.exit(message)
+        try:
+            spiceypy_cspice_path = os.path.expanduser(os.environ['SPICEYPY_CSPICE_PATH'])
+            if not os.path.isabs(spiceypy_cspice_path):
+                raise EnvironmentError("spiceypy_cspice_path must be an absolute path. Got {0}".format(
+                    spiceypy_cspice_path))
+            print("Environment variable spiceypy_cspice_path is set. Installing CSPICE from {0}.".format(
+                spiceypy_cspice_path))
+            print("Unpacking cspice.")
+            untar_cmd = "tar xzf {0} -C {1}".format(spiceypy_cspice_path, root_dir)
+            proc = subprocess.Popen(untar_cmd, shell=True)
+            proc.wait()
```

Comment:
Using a built in Python unpacking tool seems like a more robust choice here. Maybe https://docs.python.org/3/library/shutil.html#shutil.unpack_archive

Reply:
so the existing process in get_spice does use subprocess (specifically "gunzip | tar xC ") I forget the exact reason why as it has now been years but I think there is something non standard about the linux/macos distributions. 

While I thank @medley56 for the contribution, I think there are cleaner ways of implementing this functionality without changing the code so much, for example the behavior could potentially be isolated to the download function entirely with a if statement to see if the environment variable is defined and if so and the file exists pass it on as a io.ByteIO stream.

Reply:
@jessemapel I chose to use `subprocess` to match the existing behavior. I agree that this is not ideal. Also, I think there are now cleaner ways to use `subprocess` than `Popen`. I think `run` is the suggested usage now for basic cases.

@AndrewAnnex I started off modifying the `_download` and `_unpack` methods of `GetCSPICE` but decided it was not idiomatic since we are not downloading anything. The whole `GetCSPICE` class seemed very clearly directed toward the acquisition of CSPICE from NAIF and didn't feel like the right place for the new logic without renaming a bunch of stuff.

Reply:
Something like

```
try:
    Subprocess.run(["tar", "xzf", spiceypy_cspice_path, "-C", root_dir], check=True, shell=True, text=True)
except CalledProcessError as e:
    print(f"Failed to unpack CSPICE archive with return code [{e.returncode}], and error [{e.stderr}].")
```